Source/Task/TaskQueue.cpp

```diff
@@ -0,0 +1,1590 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.#include "stdafx.h"
+#include "pch.h"
+#include "referenced_ptr.h"
+#include "TaskQueueP.h"
+#include "TaskQueueImpl.h"
+
+namespace ApiDiag
```

Comment:
I think this code may run into initialization order issues.
Can any of these globals be touched as a consequence of a constructor running in another compilation unit?

Reply:
They are only present for tracking leaks in the unit tests. They aren't used at runtime.  If there are parallel copies, no harm.

Reply:
It's not about parallel copies, it's about code seeing the values before they are initialized, could you add a comment that we only use it for tests and so not care about initialization order?